mlflow_faculty/trackingstore.py

```diff
@@ -286,10 +287,32 @@ def search_runs(self, experiment_ids, search_expressions, run_view_type):
         the experiments.  Given multiple search expressions, all these
         expressions are ANDed together for search.
 
-        :param experiment_ids: List of experiment ids to scope the search
-        :param search_expression: list of search expressions
+        :param experiment_ids: List[int] of experiment ids to scope the search
+        :param search_expression: List of search expressions
 
         :return: A list of :py:class:`mlflow.entities.Run` objects that satisfy
             the search expressions
         """
-        raise NotImplementedError()
+        if search_expressions is not None:
+            raise NotImplementedError("search_expressions must be set to None")
+        if run_view_type is not None:
+            raise NotImplementedError("run_view_type must be set to None")
+
+        try:
+            faculty_runs = []
+            while True:
+                list_runs_response = self._client.list_runs(
+                    self._project_id, experiment_ids=experiment_ids
+                )
+                faculty_runs += list_runs_response.runs
+                if list_runs_response.pagination.next is None:
+                    break
```

Comment:
Should we should also break if we get zero runs back, otherwise you could imagine getting into some kind of crash loop where the next page is always the same?

Reply:
good catch, I've added a check